google-cloud-firestore/src/main/java/com/google/cloud/firestore/VectorQueryOptions.java

```diff
@@ -0,0 +1,153 @@
+     * will contain the computed distance for the document. If `null`, then the computed distance
+     * will not be returned. Default value: `null`.
+     *
+     * @param fieldPath A string value specifying the distance result field.
+     */
+    public Builder setDistanceResultField(@Nullable String fieldPath) {
+      this.distanceResultField = FieldPath.fromDotSeparatedString(fieldPath);
+      return this;
+    }
+
+    /**
+     * Specifies the name of the field that will be set on each returned DocumentSnapshot, which
+     * will contain the computed distance for the document. If `null`, then the computed distance
+     * will not be returned. Default value: `null`.
+     *
+     * @param fieldPath A {@link FieldPath} value specifying the distance result field.
+     */
+    public Builder setDistanceResultField(@Nullable FieldPath fieldPath) {
+      this.distanceResultField = fieldPath;
+      return this;
+    }
+
+    /**
+     * Specifies a threshold for which no less similar documents will be returned. The behavior of
+     * the specified `distanceMeasure` will affect the meaning of the distance threshold.
+     *
+     * <p>- For `distanceMeasure: "EUCLIDEAN"`, the meaning of `distanceThreshold` is: SELECT docs
+     * WHERE euclidean_distance <= distanceThreshold - For `distanceMeasure: "COSINE"`, the meaning
+     * of `distanceThreshold` is: SELECT docs WHERE cosine_distance <= distanceThreshold - For
+     * `distanceMeasure: "DOT_PRODUCT"`, the meaning of `distanceThreshold` is: SELECT docs WHERE
+     * dot_product_distance >= distanceThreshold
+     *
+     * <p>If `null`, then the computed distance will not be returned. Default value: `null`.
+     *
+     * @param distanceThreshold A Double value specifying the distance threshold.
+     */
+    public Builder setDistanceThreshold(@Nullable Double distanceThreshold) {
+      this.distanceThreshold = distanceThreshold;
+      return this;
+    }
+
+    public VectorQueryOptions build() {
+      return new VectorQueryOptions(this);
+    }
+  }
+
+  /**
+   * Returns true if this VectorQueryOptions is equal to the provided object.
+   *
+   * @param obj The object to compare against.
+   * @return Whether this VectorQueryOptions is equal to the provided object.
+   */
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null || !(obj instanceof VectorQueryOptions)) {
+      return false;
+    }
+    VectorQueryOptions otherOptions = (VectorQueryOptions) obj;
+    return Objects.equals(distanceResultField, otherOptions.distanceResultField)
+        && Objects.equals(distanceThreshold, otherOptions.distanceThreshold);
+  }
+
+  /**
+   * Returns a default {@code FirestoreOptions} instance. Note: package private until API review can
+   * be completed.
+   */
+  static VectorQueryOptions getDefaultInstance() {
+    return newBuilder().build();
+  }
```

Comment:
This is to avoid instantiating the same object over and over again.

```suggestion
  private static DEFAULT = newBuilder().build();

  /**
   * Returns a default {@code FirestoreOptions} instance. Note: package private until API review can
   * be completed.
   */
  static VectorQueryOptions getDefaultInstance() {
    return DEFAULT;
  }
```

Reply:
Updated.